src/app/project/project-permissions/permissions-table-rows/permissions-table-rows.component.ts

```diff
@@ -46,6 +47,33 @@ export class PermissionsTableRowsComponent implements OnInit, TableComponent {
     this.currentProject = this.storageService.state.currentProject.data;
     this.entries = this.data.data;
     this.paginationData = this.data.paginationData;
+
+    this.ngxSmartModalService.getModal('confirmation-modal').onClose.subscribe(() => {
+      const modalData = this.ngxSmartModalService.getModalData('confirmation-modal');
+      if (modalData.deleteConfirm && modalData.user._id) {
+        // If the admin selected "OK" and a valid user ID is found, remove user from project
+        this.userService.removeUser(modalData.user)
+        .takeUntil(this.ngUnsubscribe)
+        .subscribe(
+          (returnedUsers) => {
+            this.entriesVault = returnedUsers;
```

Comment:
`userService.removeUser` returns an `Observable<User>`, not an `Observable<User[]>`. If `entriesVault` is expected to be an array for `removeDuplicateUsers()`, this may cause issues. Consider setting this.entries vault to an array containing the user, like so:
```suggestion
          (returnedUser) => {
            this.entriesVault = [returnedUser];
```

Reply:
Right, so it's just a mistyped method in UserService. It should say it's returning Observable<User[]>. The find function that's performed in the API returns an array, not a single entry. The entries are all being returned, and it's working, but I'll fix the typing on the user service method.

Reply:
Quick update, there were multiple mistyped Observables in the API and the User service. They have all been updated to arrays now.

Reply:
Appreciate it!